tests/regression/71-strings/01-string_literals.c

```diff
@@ -0,0 +1,84 @@
+// PARAM: --disable ana.base.limit-string-addresses --enable ana.int.interval
+
+#include <goblint.h>
+#include <string.h>
+#include <stdlib.h>
+
+char* hello_world() {
+    return "Hello world!";
+}
+
+void id(char* s) {
+    s = s;
+}
+
+int main() {
+    char* s1 = "abcde";
+    char* s2 = "abcdfg";
+    char* s3 = hello_world();
+    
+    int i = strlen(s1);
+    __goblint_check(i == 5);
+
+    i = strlen(s2);
+    __goblint_check(i == 6);
+
+    i = strlen(s3);
+    __goblint_check(i == 12);
+
+    id(s2);
+    i = strlen(s2);
+    __goblint_check(i == 6);
+
+    i = strcmp(s1, s2);
+    __goblint_check(i < 0);
+
+    i = strcmp(s2, "abcdfg");
+    __goblint_check(i == 0);
+
+    char* cmp = strstr(s1, "bcd");
+    i = strcmp(cmp, "bcde");
+    __goblint_check(i == 0);
+
+    cmp = strstr(s1, "bcdf");
+    __goblint_check(cmp == NULL);
+
+    if (rand() == 42)
+        s3 = "hello";
+    else
+        s3 = "world";
+
+    cmp = strstr(s3, "l");
+    __goblint_check(cmp != NULL);
+
+    cmp = strstr(s3, "he");
+    __goblint_check(cmp != NULL); // UNKNOWN
+
+    i = strncmp(s1, s2, 4);
+    __goblint_check(i == 0);
+
+    i = strncmp(s1, s2, 5);
+    __goblint_check(i != 0);
+    
+    #ifdef __APPLE__
+        /* the following portion fails on macOS because of a spurious warning:
+         * see issue goblint/cil#143
+         *
+         * remove #ifdef portion and change "should warn" to normal warning as soon as issue fixed */
+    #else
+        strcpy(s1, "hi"); // should warn
+        strncpy(s1, "hi", 1); // should warn
+        strcat(s1, "hi"); // should warn
+        strncat(s1, "hi", 1); // should warn
```

Comment:
If they should warn, then it should be `// WARN`. Otherwise this isn't really testing anything.

Reply:
The problem is that if I keep `// WARN`, than the test still fails on macOS because no warning was issued as the portion is never reached on an Apple device. Is there any smart way to test if the warnings are correctly generated on Linux while ignoring that annotation on OS X?

Reply:
We decided to go with `*0;` in the OS X case so we get a warning. This is hacky, but I guess ok.

Reply:
Nice that you managed to make it work!